lib/rollbax/item.ex

```diff
@@ -2,33 +2,76 @@ defmodule Rollbax.Item do
   def draft(token, envt) do
     {:ok, host} = :inet.gethostname
     %{
-      "access_token" => token,
-      "data" => %{
-        "server" => %{
-          "host" => List.to_string(host)
+      access_token: token,
+      data: %{
+        server: %{
+          host: List.to_string(host)
         },
-        "environment" => envt,
-        "language" => language(),
-        "platform" => platform(),
-        "notifier" => notifier()
+        environment: envt,
+        language: language(),
+        platform: platform(),
+        notifier: notifier()
       }
     }
   end
 
-  def compose(draft, {level, msg, time, meta}) do
+  def compose(draft, {level, e, stacktrace, time, meta}) do
     {occurr_data, meta} =
       Map.pop(meta, :rollbax_occurr_data, %{})
-    Map.update!(draft, "data", fn(data) ->
+    Map.update!(draft, :data, fn(data) ->
       Map.merge(occurr_data, data)
-      |> put_body(msg)
+      |> put_body(e, stacktrace)
       |> put_custom(meta)
-      |> Map.put("level", level)
-      |> Map.put("timestamp", time)
+      |> Map.put(:level, level)
+      |> Map.put(:timestamp, time)
     end)
   end
 
-  defp put_body(data, msg) do
-    Map.put(data, "body", %{"message" => %{"body" => msg}})
+  # If a message with no stack trace, use "message"
+  defp put_body(data, e, nil) do
+    Map.put(data, :body, %{ message: %{ body: Exception.format(:error, e, nil) } })
+  end
+
+  # If this payload is a single exception, use "trace"
+  # stacktrace example: [{Test, :report, 2, [file: 'file.exs', line: 16]}]
+  defp put_body(data, e, stacktrace) do
+    Map.put(data, :body, %{ trace: trace(e, stacktrace) })
+  end
+
+  defp trace(e, stacktrace) do
+    %{ frames: frames(stacktrace), exception: exception(e) }
+  end
+
+  # Required: frames
+  # A list of stack frames, ordered such that the most recent call is last in the list.
+  # TODO implement all options. See https://rollbar.com/docs/api/items_post/
+  defp frames(stacktrace) do
+    Enum.map(stacktrace, fn({_, _, _, file}) ->
+      %{
+        # Required: filename
+        # The filename including its full path
+        filename: file[:file],
+
+        # Optional: lineno
+        # The line number as an integer
+        lineno: file[:line]
+      }
+    end)
```

Comment:
I think it would be better to avoid adding the `lineno` and `filename` keys in the frame if they're not there (smaller payload, lower chance of messing with Rollbar and how it treats nulls).